include/onnxruntime/core/graph/graph.h

```diff
@@ -1519,6 +1489,28 @@ class Graph {  // NOLINT(clang-analyzer-optin.performance.Padding): preserve exi
   Status AddConstantProtoAsInitializer(const ONNX_NAMESPACE::NodeProto& constant_node_proto,
                                        std::optional<std::string_view> new_name);
 
+  /// <summary>
+  /// A map that is used to keep track of pre-packed blobs to be serialized
+  /// The implementation adds pre-packed external data references to the TensorProto
+  /// that contains the initializer data. However, it may be an outerscope initializer.
+  /// Thus we need to keep track of the pre-packed blobs that are not serialized in this
+  /// graph, so the parent can make sure it is being serialized.
```

Comment:
Can we just keep the prepacked weight in the same Graph instance as the initializer? Seems complicated to be doing this sort of tracking vs. keeping the original and prepacked values in the same Graph instance. 

Maybe that needs a similar lookup logic as we have for initializers with an outerscope search, but that's hopefully relatively trivial to implement.